src/sentry/auth/helper.py

```diff
@@ -466,6 +488,12 @@ def handle_unknown_identity(
         else:
             op = None
 
+        if self.request.session.get("confirm_account_verification_key"):
```

Comment:
I think this lives best in in the above control statement, above the line ` elif acting_user and not acting_user.has_usable_password():`